session.go

```diff
@@ -374,7 +374,7 @@ func (s *Session) send() {
 				for sent < len(ready.Hdr) {
 					n, err := s.conn.Write(ready.Hdr[sent:])
 					if err != nil {
-						s.logger.Printf("[ERR] yamux: Failed to write header: %v", err)
+						s.logger.Printf("[WARN] yamux: Failed to write header: %v", err)
```

Comment:
Why exactly should it be a warning and not an error? If I understand correctly this will completely stop the yamux connection. I would think with just a warning things would be able to recover and continue to work, not stop completely.

Reply:
@erikdubbelboer none of those additional commits should have appeared in this PR. We are maintaining our own fork.

---

An `error` is still returned to the client, this only changes a _log_ message. I consider yamux consistency issues bugs and should be logged as **errors**. An example of a consistency issue would  `yamux: SYN tracking out of sync` and highlights a bug in yamux. The network connection being closed, causing a write to fail is not a yamux bug and is still reported as an `error` to the client code, however it generates unnecessary log noise.

Reply:
In that case should it log anything at all? Maybe it should just return the error and let the application log something if it wants. I always thing libraries logging things if only for when it can't tell its user that something happened (by returning an error for example).

Reply:
I definitely agree. The logging within the yamux package should be restricted to yamux internal consistency issues. Logging a warning because the network connection has been closed is redundant, since the error will be passed to the caller